internal/monitors/expvar/expvar.go

```diff
@@ -0,0 +1,393 @@
+	if _map, ok := v.(map[string]interface{}); ok {
+		for _, dConf := range mConf.DimensionConfigs {
+			if len(dConf.splits) != 0 && len(dConf.splits) == i {
+				dp.Dimensions[dConf.Name] = mConf.keys[i]
+			}
+		}
+		setDatapoints(_map[mConf.keys[i+1]], mConf, dp, dpsMap, i+1)
+	} else if arr, ok := v.([]interface{}); ok {
+		_dp := dp
+		for _i, _v := range arr {
+			if _i > 0 {
+				// At this point nothing is set for the do except possibly some dimensions
+				_dp = &datapoint.Datapoint{Dimensions: copyDims(dp)}
+			}
+			makeNewIndexDimension := true
+			for _, dConf := range mConf.DimensionConfigs {
+				if len(dConf.splits) == i+1 {
+					_dp.Dimensions[dConf.Name] = strconv.Itoa(_i)
+					makeNewIndexDimension = false
+				}
+			}
+			if makeNewIndexDimension {
+				_dp.Dimensions[strings.Join(mConf.keys[:(i+1)], ".")] = strconv.Itoa(_i)
+			}
+			setDatapoints(_v, mConf, _dp, dpsMap, i)
+		}
+	} else {
+		if v == nil {
+			logger.Errorf("failed to find value for metric %s with JSON path %s", mConf.Name, mConf.JSONPath)
+			return
+		}
+		dp.Metric, dp.MetricType = mConf.Name, mConf.metricType
+		for _, dConf := range mConf.DimensionConfigs {
+			if strings.TrimSpace(dConf.Name) != "" && strings.TrimSpace(dConf.Value) != "" {
+				dp.Dimensions[dConf.Name] = dConf.Value
+			}
+		}
+		var err error
+		if dp.Value, err = datapoint.CastMetricValueWithBool(v); err == nil {
+			(*dpsMap)[mConf.JSONPath] = append((*dpsMap)[mConf.JSONPath], dp)
+		} else {
+			logger.Errorf("failed to set value for metric %s with JSON path %s because of type conversion error due to %+v", mConf.Name, mConf.JSONPath, err)
+			return
+		}
+	}
+}
+
+// getMostRecentGCPauseIndex logic is derived from https://golang.org/pkg/runtime/ in the PauseNs section of the 'type MemStats' section
+func getMostRecentGCPauseIndex(dpsMap *map[string][]*datapoint.Datapoint) int64 {
+	dps := (*dpsMap)[memstatsNumGCMetricPath]
+	mostRecentGCPauseIndex := int64(-1)
+	if len(dps) > 0 && dps[0].Value != nil {
+		if numGC, err := strconv.ParseInt(dps[0].Value.String(), 10, 0); err == nil {
+			mostRecentGCPauseIndex = (numGC + 255) % 256
+		}
+	}
+	return mostRecentGCPauseIndex
+}
+
+var slashLastRegexp = regexp.MustCompile("[^\\/]*$")
+
+func getApplicationName(_map map[string]interface{}) (string, error) {
+	if cmdline, ok := _map["cmdline"].([]interface{}); ok && len(cmdline) > 0 {
+		if applicationName := strings.TrimSpace(slashLastRegexp.FindStringSubmatch(cmdline[0].(string))[0]); applicationName != "" {
+			return applicationName, nil
+		}
+	}
+	return "", fmt.Errorf("failed to get application name from the first array value of cmdline in map: %+v", _map)
+}
+
+func copyDims(dp *datapoint.Datapoint) map[string]string {
+	dimensions := map[string]string{}
+	for k, v := range dp.Dimensions {
+		dimensions[k] = v
+	}
+	return dimensions
+}
+
+var camelRegexp = regexp.MustCompile("(^[^A-Z]*|[A-Z]*)([A-Z][^A-Z]+|$)")
+
+func toSnakeCase(s string) string {
```

Comment:
This is already implemented in a library that the agent already depends on: `strcase.ToSnake`.